asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+    df = df.rename(columns={new_col:"SQ45"})
+    df_45 = pd.concat([df_45, df])
+
+# Generate crosstab dataframe
+df_607_crosstab = crosstable("EmploymentType",
+                             df_45,
+                             "SQ45",
+                             ans_45
+                             )
+df_607_crosstab = df_607_crosstab.fillna(0)
+df_607_crosstab = df_607_crosstab.astype(int)
+
+# Remove Employee index row because they are not asked
+df_607_crosstab = df_607_crosstab.drop(index='Employee')
+
+# Save to .csv and display
+df_607_crosstab.to_csv("../../outputs/section6/csv/recoded/607_recoded.csv")
+df_607_crosstab
+
+# %%
+# Generate figure
+groupedbar(df_607_crosstab,
+           "607_recoded",
+           "SQ 45. Have you found any challenges working in heat pump repairs, servicing and maintenance?")
+
+# %%
+# Generate figure
+donut(df_607_crosstab,
+           "607_recoded",
+           "SQ 45. Have you found any challenges working in heat pump repairs, servicing and maintenance?")
+
+# %% [markdown]
+# ### RQ 6.08 | How do the challenges that survey participants find with working in repairs, servicing and maintenance differ depending on the size of the company they own?
+# - SQ: 45
+# - Sample:
+# - Company_size_owner
+
+# %%
+# CompanySizeOwner V2
+# Create separate dataframe that stacks all answer columns
+df_p = pd.DataFrame()
+for n in range(0, len(ans_45) - 1):
+
+    # Create new column for each answer option
+    response = ans_45[n]
+    new_col = "SQ45Ans"+str(n + 1)
+    data[new_col] = data.apply(transform_45, axis=1)
+    
+    # Stack columns
+    df = data.filter(["CompanySizeOwnerV2", new_col])
+    df = df.rename(columns={new_col:"SQ45"})
+    df_p = pd.concat([df_p, df])
+
+# Generate crosstab dataframe
+df_608_crosstab_owner_V2 = crosstable("CompanySizeOwnerV2",
+                                      df_p,
+                                     "SQ45",
+                                     ans_45
+                                     )
+df_608_crosstab_owner_V2 = df_608_crosstab_owner_V2.fillna(0)
+df_608_crosstab_owner_V2 = df_608_crosstab_owner_V2.astype(int)
+
+# Save to .csv and display
+df_608_crosstab_owner_V2.to_csv("../../outputs/section6/csv/recoded/608_owner_V2_recoded.csv")
+df_608_crosstab_owner_V2
+
+# %%
+# Generate figure
+groupedbar(df_608_crosstab_owner_V2,
+           "608_owner_V2_recoded",
+           "SQ 45. Have you found any challenges working in heat pump repairs, servicing and maintenance?")
+
+# %%
+# Generate figure
+donut(df_608_crosstab_owner_V2,
+           "608_owner_V2_recoded",
+           "SQ 45. Have you found any challenges working in heat pump repairs, servicing and maintenance?")
+
+# %% [markdown]
+# ### RQ 6.09 | Do survey participants who currently work in repairs,servicing and maintenance expect to expand this activity?
+# - SQ: 46
+# - Sample:
+# - Standard_output
+
+# %%
+# Print possible answers
+data[col.q46].value_counts()
```

Comment:
More of a curiosity, but how come so many were not asked for this question?

Reply:
According to the [survey logic](https://miro.com/app/board/uXjVNWK_xBg=/), SQ 46 only appeared for those who said "Yes" to SQ 42a "Do you offer repairs, servicing and maintenance ..." (and only company owners, sole traders and contractors were asked SQ 42a)